chain/network/src/peer_manager/tests.rs

```diff
@@ -156,6 +166,67 @@ async fn no_edge_broadcast_after_restart() {
     }
 }
 
+// Nonces must be odd (as even ones are reserved for tombstones).
+fn to_active_nonce(timestamp: time::Utc) -> u64 {
+    let value = timestamp.unix_timestamp() as u64;
+    if value % 2 == 0 {
+        value + 1
+    } else {
+        value
+    }
+}
+
+// Test connecting to peer manager with timestamp-like nonces.
+#[tokio::test]
+async fn test_nonces() {
+    init_test_logger();
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::new(*EDGE_MIN_TIMESTAMP_NONCE + time::Duration::days(2));
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    //let mut total_edges = vec![];
+    let store = create_test_store();
+
+    // Start a PeerManager and connect a peer to it.
+    let pm = peer_manager::testonly::start(
+        clock.clock(),
+        store.clone(),
+        chain.make_config(rng),
+        chain.clone(),
+    )
+    .await;
+
+    let test_cases = [
+        // Try to connect with peer with a valid nonce (current timestamp).
+        (Some(to_active_nonce(clock.now_utc())), true),
+        // Now try the peer with invalid timestamp (in the past)
+        (Some(to_active_nonce(clock.now_utc() - time::Duration::days(1))), false),
+        // Now try the peer with invalid timestamp (in the future)
+        (Some(to_active_nonce(clock.now_utc() + time::Duration::days(1))), false),
+    ];
+
+    for test in test_cases {
+        let cfg = peer::testonly::PeerConfig {
+            signer: data::make_signer(rng),
+            chain: chain.clone(),
+            peers: vec![],
+            start_handshake_with: Some(PeerId::new(pm.cfg.node_key.public_key())),
+            force_encoding: Some(Encoding::Proto),
+            // Connect with nonce equal to unix timestamp
+            nonce: test.0,
+        };
+        let stream = TcpStream::connect(pm.cfg.node_addr.unwrap()).await.unwrap();
+        let mut peer =
+            peer::testonly::PeerHandle::start_endpoint(clock.clock(), rng, cfg, stream).await;
+        if test.1 {
+            peer.complete_handshake().await;
+        } else {
+            peer.fail_handshake().await;
+        }
```

Comment:
since we are running test cases in a loop, the logs won't indicate in which test case the error happened.
You should indicate that somehow. Here the easiest way would be to print a log at the beginning of each test case.

Reply:
good idea - done.